test/libsolidity/SemanticTest.cpp

```diff
@@ -353,11 +380,24 @@ void SemanticTest::printUpdatedSettings(ostream& _stream, string const& _linePre
 		return;
 
 	_stream << _linePrefix << "// ====" << endl;
-	if (m_compileViaYulCanBeSet)
+	if (m_compileToEwasmCanBeSet)
+	{
+		if (m_runWithYul)
+		{
+			if (m_runWithoutYul)
+				_stream << _linePrefix << "// compileViaYul: also\n";
+			else
+				_stream << _linePrefix << "// compileViaYul: true\n";
+		}
+
+		_stream << _linePrefix << "// compileToEwasm: also\n";
+	}
+	else if (m_compileViaYulCanBeSet)
```

Comment:
I find this block a bit confusing with all these flags. Why can't `m_compileViaYulCanBeSet` be used above the same way it's used here \/ for `also`?

Reply:
I added some comments to this. The reason is that `m_compileViaYulCanBeSet` is only set, if a test can be compiled via yul (and it was not configured to run via yul before). That means, it is not set if the test is already configured as `compileViaYul: also` or `compileViaYul: true`. If a test can be compiled to ewasm we need to rewrite these settings.

Reply:
Ah ok

Reply:
I think it would be clearer if this was checking the value of `m_reader.stringSetting("compileViaYul", "default")` instead.